src/coreclr/vm/jithelpers.cpp

```diff
@@ -561,6 +561,46 @@ HCIMPL1_V(float, JIT_FloatRound, float val)
 }
 HCIMPLEND
 
+#ifndef TARGET_WINDOWS
+namespace
+{
+    bool isnan(float val)
+    {
+        UINT32 bits = *reinterpret_cast<UINT32*>(&val);
+        return (bits & 0x7FFFFFFFU) > 0x7F800000U;
+    }
+    bool isnan(double val)
+    {
+        UINT64 bits = *reinterpret_cast<UINT64*>(&val);
+        return (bits & 0x7FFFFFFFFFFFFFFFULL) > 0x7FF0000000000000ULL;
+    }
+    bool isfinite(float val)
+    {
+        UINT32 bits = *reinterpret_cast<UINT32*>(&val);
+        return (~bits & 0x7F800000U) != 0;
+    }
+    bool isfinite(double val)
+    {
+        UINT64 bits = *reinterpret_cast<UINT64*>(&val);
+        return (~bits & 0x7FF0000000000000ULL) != 0;
+    }
+}
+#endif
+
+/*********************************************************************/
+HCIMPL1_V(INT64, JIT_Dbl2Lng, double val)
+{
+    FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    const double int64_min = -2147483648.0 * 4294967296.0;
+    const double int64_max = 2147483648.0 * 4294967296.0;
+    return (val!= val) ? 0 : (val <= int64_min) ? INT64_MIN : (val >= int64_max) ? INT64_MAX : (INT64)val;
+#else
+    return((INT64)val);
+#endif // TARGET_X86 || TARGET_AMD64
+}
+HCIMPLEND
+
 /*********************************************************************/
 // Call fast Dbl2Lng conversion - used by functions below
 FORCEINLINE INT64 FastDbl2Lng(double val)
```

Comment:
Does this serve any purpose? Should this be deleted and the conversion inlined in the few places that still call it?

Reply:
It was not inlined because ARM still calls into this helper function
https://github.com/khushal1996/runtime/blob/f3e4bf5556f730dec35c8951aa97ba9480f5413b/src/coreclr/jit/morph.cpp#L617

Reply:
I meant manually inline it into the other C helpers. I did not mean inlining in the JIT.

Reply:
For example, make `JIT_Dbl2ULng` implementation to be identical to `RhpDbl2ULng`

Reply:
Are you talking about ```FastDbl2Lng``` or ```JIT_Dbl2Lng```? 

Also,
The implementation of ```JIT_Dbl2ULng``` is identical to that of ```RhpDbl2ULng```. Can you elaborate if you are talking about the implementation in this PR which are identical or something else?

Reply:
My bad. I think you are talking about ```FastDbl2Lng```. I think it should be kept as is since it is used in ```JIT_Dbl2ULng```, ```JIT_Dbl2UIntOvf``` and ```JIT_Dbl2ULngOvf```. With the overflow conversions being used by all architecures and difference between implementation of ```JIT_Dbl2Lng```, inlining things would make it more complicated to read and understand.